src/GitHub.App/Services/PullRequestService.cs

```diff
@@ -95,5 +255,33 @@ public IObservable<string> GetPullRequestTemplate(ILocalRepositoryModel reposito
             return ret;
         }
 
+        void EnsurePullRefSpecExists(IRepository repo)
+        {
+            var spec = "+refs/pull/*/head:refs/remotes/origin/pr/*";
+            var origin = repo.Network.Remotes["origin"];
+            var existing = origin.FetchRefSpecs.FirstOrDefault(x => x.Specification == spec);
+
+            if (existing == null)
+            {
+                repo.Network.Remotes.Update(origin, x => x.FetchRefSpecs.Add(spec));
+            }
+        }
+
+        static string GetSafeBranchName(string name)
```

Comment:
Hmm I guess someone *could* create a PR using only punctuation and no letters or digits, though I'm not sure how likely it would be! And yeah, in that case the branch name would be `PR/123--`.